src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -137,6 +149,64 @@ void AndroidDeviceControllerWrapper::OnPairingDeleted(CHIP_ERROR error)
     CallJavaMethod("onPairingDeleted", static_cast<jint>(error));
 }
 
+CHIP_ERROR AndroidDeviceControllerWrapper::GenerateNodeOperationalCertificate(const chip::PeerId & peerId, const chip::ByteSpan & csr,
+                                                                                int64_t serialNumber, uint8_t * certBuf,
+                                                                                uint32_t certBufSize, uint32_t & outCertLen)
+{
+    jmethodID method;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    err = FindMethod(GetEnvForCurrentThread(), mJavaObjectRef, "onOpCSRGenerationComplete", "([B)V", &method);
+    if (err != CHIP_NO_ERROR)
+   {
+      ChipLogError(Controller, "Error invoking onOpCSRGenerationComplete: %d", err);
+      return err;
+   }
+
+    // Initializing the KeyPair.
+     Initialize();
+
+    chip::X509CertRequestParams request = { serialNumber, mIssuerId,         mNow, mNow + mValidity, true, peerId.GetFabricId(),
+                                          true,         peerId.GetNodeId() };
+
+    chip::P256PublicKey pubkey;
+    ReturnErrorOnFailure(VerifyCertificateSigningRequest(csr.data(), csr.size(), pubkey));
+
+    ChipLogProgress(chipTool, "VerifyCertificateSigningRequest");
+
+
+    CHIP_ERROR generateCert = NewNodeOperationalX509Cert(request, chip::CertificateIssuerLevel::kIssuerIsRootCA, pubkey, mIssuer, certBuf, certBufSize,
+                                         outCertLen);
+    jbyteArray argument;
+    GetEnvForCurrentThread()->ExceptionClear();
+    N2J_ByteArray(GetEnvForCurrentThread(), csr.data(),csr.size(),argument);
+    GetEnvForCurrentThread()->CallVoidMethod(mJavaObjectRef, method, argument);
```

Comment:
What if generateCert failed? is this still OK to be called?

Reply:
Currently, the method CHIPCert.NewNodeOperationalX509Cert  will generate a CHIP_Error if the cert generation fails and so will GenerateNodeOperationalCertificate. But the whole relevance of adding this method here for my use case is to trigger the generation of OpCSR using the passed nonce. The generated cert is not useful. Ideally the generated CSRNonce will be exchanged to get an OpCert which needs to be written to the device in a different api call (after the PairDevice completes and that api does not exist today). 

It would have been ideal to have a separate api to just pass the CSRNonce and get the CSRResponse but with current sdk support this is the only possible path. 

Lemme know if this makes sense.